elixir/apps/domain/lib/domain/billing/stripe/processed_events/event.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Domain.Billing.Stripe.ProcessedEvent do
+  @moduledoc """
+  Schema for tracking processed Stripe webhook events.
+  Ensures idempotency and chronological ordering of event processing.
+  """
+
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key {:stripe_event_id, :string, []}
+  schema "processed_stripe_events" do
+    field :event_type, :string
+    field :processed_at, :utc_datetime
+    field :stripe_customer_id, :string
+    field :event_created_at, :utc_datetime
+    field :livemode, :boolean, default: false
+
+    timestamps()
+  end
+
+  @required_fields [
+    :stripe_event_id,
+    :event_type,
+    :processed_at,
+    :event_created_at,
+    :livemode
+  ]
+
+  @optional_fields [
+    :stripe_customer_id
+  ]
+
+  @doc false
+  def changeset(processed_event, attrs) do
```

Comment:
I know it's a bit overkill sometimes but it might be better to stick to our convention of using different modules for schema and changeset.

Reply:
Yeah, probably a good idea.  I'll updated it.